cardano-api/src/Cardano/Api/DeserialiseAnyOf.hs

```diff
@@ -0,0 +1,322 @@
+          textEnvRes =
+            deserialiseFromTextEnvelopeAnyOf textEnvTypes
+              =<< first TextEnvelopeAesonDecodeError (Aeson.eitherDecodeStrict' inputBs)
+      case textEnvRes of
+        Right res -> DeserialiseInputSuccess res
+
+        -- The input was valid a text envelope, but there was a type mismatch
+        -- error.
+        Left err@TextEnvelopeTypeError{} ->
+          DeserialiseInputError (InputTextEnvelopeError err)
+
+        -- The input was not valid a text envelope.
+        Left _ -> DeserialiseInputErrorFormatMismatch
+
+    deserialiseBech32 :: DeserialiseInputResult b
+    deserialiseBech32 =
+      case deserialiseAnyOfFromBech32 bech32Types inputText of
+        Right res -> DeserialiseInputSuccess res
+
+        -- The input was not valid Bech32.
+        Left (Bech32DecodingError _) -> DeserialiseInputErrorFormatMismatch
+
+        -- The input was valid Bech32, but some other error occurred.
+        Left err -> DeserialiseInputError $ InputBech32DecodeError err
+
+data SomeAddressVerificationKey
+  = AByronVerificationKey           (VerificationKey ByronKey)
+  | APaymentVerificationKey         (VerificationKey PaymentKey)
+  | APaymentExtendedVerificationKey (VerificationKey PaymentExtendedKey)
+  | AGenesisUTxOVerificationKey     (VerificationKey GenesisUTxOKey)
+  | AGenesisExtendedVerificationKey (VerificationKey GenesisExtendedKey)
+  | AGenesisDelegateExtendedVerificationKey
+                                    (VerificationKey GenesisDelegateExtendedKey)
+  | AKesVerificationKey             (VerificationKey KesKey)
+  | AVrfVerificationKey             (VerificationKey VrfKey)
+  | AStakeVerificationKey           (VerificationKey StakeKey)
+  | AStakeExtendedVerificationKey   (VerificationKey StakeExtendedKey)
+  deriving (Show)
+
+
+renderSomeAddressVerificationKey :: SomeAddressVerificationKey -> Text
+renderSomeAddressVerificationKey (AByronVerificationKey vk) =  prettyByronVerificationKey vk
+renderSomeAddressVerificationKey (APaymentVerificationKey vk) = serialiseToBech32 vk
+renderSomeAddressVerificationKey (APaymentExtendedVerificationKey vk) = serialiseToBech32 vk
+renderSomeAddressVerificationKey (AGenesisUTxOVerificationKey vk) =
+  serialiseToBech32 (castVerificationKey vk :: VerificationKey PaymentKey)
+renderSomeAddressVerificationKey (AGenesisExtendedVerificationKey vk) =
+  let genKey =  (castVerificationKey vk :: VerificationKey GenesisKey)
+      payKey = (castVerificationKey genKey :: VerificationKey PaymentKey)
+  in serialiseToBech32 payKey
+renderSomeAddressVerificationKey (AGenesisDelegateExtendedVerificationKey vk) =
+  -- TODO: We could implement a CastVerificationKeyRole GenesisDelegateKey PaymentKey
+  -- if we want to avoid casting twice.
+  let genDelegKey = (castVerificationKey vk :: VerificationKey GenesisDelegateKey)
+      stakePoolKey = castVerificationKey genDelegKey :: VerificationKey StakePoolKey
+  in serialiseToBech32 stakePoolKey
```

Comment:
Has the comment become outdated? I don't think you want `CastVerificationKeyRole GenesisDelegateKey PaymentKey`. It looks to me like you want `CastVerificationKeyRole GenesisKey PaymentKey` and `CastVerificationKeyRole GenesisDelegateKey StatePoolKey`